pkg/service/admission/externalip_admission.go

```diff
@@ -0,0 +1,62 @@
+package admission
+
+import (
+	"fmt"
+	"io"
+	"net"
+
+	kadmission "k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+)
+
+const ExternalIPPluginName = "ExternalIPRanger"
+
+func init() {
+	kadmission.RegisterPlugin("ExternalIPRanger", func(client clientset.Interface, config io.Reader) (kadmission.Interface, error) {
+		return NewExternalIPRanger(nil), nil
+	})
+}
+
+type externalIPRanger struct {
+	*kadmission.Handler
+	network *net.IPNet
+}
+
+var _ kadmission.Interface = &externalIPRanger{}
+
+// NewConstraint creates a new SCC constraint admission plugin.
+func NewExternalIPRanger(network *net.IPNet) *externalIPRanger {
+	return &externalIPRanger{
+		Handler: kadmission.NewHandler(kadmission.Create, kadmission.Update),
+		network: network,
+	}
+}
+
+// Admit determines if the service should be admitted based on the configured network CIDR.
+func (r *externalIPRanger) Admit(a kadmission.Attributes) error {
+	if a.GetResource() != kapi.Resource("services") {
+		return nil
+	}
+
+	svc, ok := a.GetObject().(*kapi.Service)
+	// if we can't convert then we don't handle this object so just return
+	if !ok {
+		return nil
+	}
+
+	switch {
+	// administrator disabled externalIPs
+	case len(svc.Spec.ExternalIPs) > 0 && r.network == nil:
+		return kadmission.NewForbidden(a, fmt.Errorf("externalIPs have been disabled and may not be set on services"))
+	// administrator has limited the range
+	case len(svc.Spec.ExternalIPs) > 0 && r.network != nil:
+		for _, s := range svc.Spec.ExternalIPs {
+			ip := net.ParseIP(s)
+			if ip == nil || !r.network.Contains(ip) {
+				return kadmission.NewForbidden(a, fmt.Errorf("externalIPs must be in the range %s", r.network))
```

Comment:
Check them all and give me the indexes and values like we do for validation?  It would make the rejection easier to interpret, this is more like a configurable validation rule than anything else.